homeassistant/components/ambient_station/__init__.py

```diff
@@ -93,7 +93,7 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
     if unload_ok:
         ambient = hass.data[DOMAIN].pop(entry.entry_id)
-        hass.async_create_task(ambient.ws_disconnect())
+        hass.async_create_task(ambient.ws_disconnect(), eager_start=True)
```

Comment:
Why do we create a task if we can start it eagerly?

Reply:
I think it could be awaited here, but since we don't need the disconnect to finish before a reload starts another connection I didn't change it to await as it would slightly delay a reload.

Reply:
Ok.